### PR TITLE
Move the module checking and updating to lib/python

### LINKED ISSUE #12413: [Bug] CLI can't cope with renamed packages in requirements.txt
## Describe the Bug

`qmk` CLI now asks if modules should be installed if not present. When installing `pillow`, it needs to check against `PIL`, which is reflected in `bin/qmk`. However, with movement towards using global CLI as much as possible, a solution needs to be available for PRs and branches to specify new entries in requirements.txt that have naming overrides. Current implementation requires removal of checks in the installed global CLI in order to execute anything using `qmk`.

image

## System Information
Debian 10, python 3.7.3.

### PR DESCRIPTION
**Pull request by skullydazed**
## Types of Changes

- [x] Core
- [x] Enhancement/optimization

## Issues Fixed or Closed by This PR

* fixes #12413 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by tzarc — APPROVED** [review]
Weak approval.

File: lib/python/qmk/cli/__init__.py [hunk 1]
```diff
@@ -49,3 +96,48 @@
...
+
+if cli.config.user.developer and _broken_module_imports('requirements-dev.txt'):
+    if yesno('Would you like to install the required developer Python modules?'):
+        _run_cmd(sys.executable, '-m', 'pip', 'install', '-r', 'requirements.txt')
+    else:
+        print()
+        print(msg_install % (str(Path('requirements-dev.txt').resolve()),))
+        print('You can also turn off developer mode: qmk config user.developer=None')
+        print()
+        exit(1)
```
Comment (Erovia) [inline lib/python/qmk/cli/__init__.py:120]:
```suggestion
    if len(sys.argv) == 1 or (len(sys.argv) > 1 and 'config' != sys.argv[1]):
        if yesno('Would you like to install the required developer Python modules?'):
            _run_cmd(sys.executable, '-m', 'pip', 'install', '-r', 'requirements-dev.txt')
        else:
            print()
            print(msg_install % (str(Path('requirements-dev.txt').resolve()),))
            print('You can also turn off developer mode: qmk config user.developer=None')
            print()
            exit(1)
```
We want to allow the user to run `qmk config` so they can turn off dev mode.

**Review by Erovia — APPROVED** [review]

**Review by Erovia — CHANGES_REQUESTED** [review]
Misclicked... -.-

File: lib/python/qmk/cli/__init__.py (hunk 1 above)
Comment (Erovia) [inline lib/python/qmk/cli/__init__.py:120]:
```suggestion
if cli.config.user.developer:
    args = sys.argv[1:]
    while args and args[0][0] == '-':
        del args[0]
    if not args or args[0] != 'config':
        if _broken_module_imports('requirements-dev.txt'):
            if yesno('Would you like to install the required developer Python modules?'):
                _run_cmd(sys.executable, '-m', 'pip', 'install', '-r', 'requirements-dev.txt')
            else:
                print()
                print(msg_install % (str(Path('requirements-dev.txt').resolve()),))
                print('You can also turn off developer mode: qmk config user.developer=None')
                print()
                exit(1)
```
We want to allow the user to run `qmk config` so they can turn off dev mode.

**Review by Erovia — APPROVED** [review]

**Pull request merged**